app/src/org/commcare/dalvik/activities/CommCareSetupActivity.java

```diff
@@ -244,7 +246,19 @@ private void uiStateScreenTransition() {
                 break;
 
             default:
-                fragment = installFragment;
+                List<Fragment> fgmts = fm.getFragments();
+                int lastIndex = fgmts != null ? fgmts.size() - 1 : -1;
+                if(lastIndex > -1) {
+                    fragment = fgmts.get(lastIndex);
+                    if (BuildConfig.DEBUG) {
+                        Log.v(TAG, "Last fragment: " + fragment);
+                    }
+                }
+                if(!(fragment instanceof SetupEnterURLFragment)){
+                    fragment = installFragment;
+                } else {
+                    // we restore the state of the SetupEnterURLFragment if it is our last fragment

```

Comment:
Please avoid empty conditional bodies. You can change the comment to not need them:
```
if(!(fragment instanceof SetupEnterURLFragment)){
    // last fragment wasn't url entry, so default to the installation method chooser
    fragment = installFragment;
}
```
My comment in the example above is a little over the top, you should really be able to use var names that self-document what is going on.